feature/login/src/main/java/com/epmedu/animeal/feature/login/code/CodeConfirmationViewModel.kt

```diff
@@ -0,0 +1,94 @@
+package com.epmedu.animeal.feature.login.code
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+internal class CodeConfirmationViewModel : ViewModel() {
+
+    private val _state = MutableStateFlow(
+        CodeConfirmationScreenState(
+            phoneNumber = getPhoneNumber(),
+            code = getEmptyCode(),
+        )
+    )
+    val state: StateFlow<CodeConfirmationScreenState> = _state.asStateFlow()
+
+    private val currentState get() = _state.value
+
+    init {
+        viewModelScope.launch {
+            launchResendTimer()
+        }
+    }
+
+    private fun getPhoneNumber(): String {
+        // TODO: Retrieve a phone number from repository
+        return PHONE_NUMBER_PLACEHOLDER
+    }
+
+    private fun getEmptyCode() = List(CODE_SIZE) { null }
+
+    private suspend fun launchResendTimer() {
+        for (tick in RESEND_DELAY downTo 1L) {
```

Comment:
Why do we need this timer here?

Reply:
Do you want to move it in composable or you want to suggest better approach to implement the timer?